quinn/src/endpoint.rs

```diff
@@ -805,3 +855,16 @@ impl std::ops::Deref for EndpointRef {
         &self.0
     }
 }
+
+struct ReceiveContext<'a, 'b: 'c, 'c> {
+    cx: &'c mut Context<'b>,
+    now: Instant,
+    iovs: &'a mut [IoSliceMut<'a>; BATCH_SIZE],
+    metas: &'c mut [RecvMeta; BATCH_SIZE],
+}
+
+struct PollResult {
+    received_data: bool,
+    keep_going: bool,
+    err: Option<io::Error>,
```

Comment:
Nit: in Rust, it would be more idiomatic to call this `error` (we usually write out field and type names).